### PR TITLE
Improve Find Command

### PR DESCRIPTION
**Pull request by guokweijie**

### CODE REVIEW

**Review by Fuzanwenn — APPROVED** [review]
LGTM

**Review by TZL0 — CHANGES_REQUESTED** [review]
Is it better to combine FindCommand and FindTagCommand into FindPatientCommand

File: src/main/java/seedu/address/model/patient/NameContainsKeywordsPredicatePatient.java [hunk 1]
```diff
@@ -1,31 +1,27 @@
...
 import java.util.function.Predicate;
 
-import seedu.address.commons.util.StringUtil;
-
 /**
  * Tests that a {@code Patient}'s {@code Name} matches any of the keywords given.
  */
 public class NameContainsKeywordsPredicatePatient implements Predicate<Patient> {
-    private final List<String> keywords;
+    private final String predicateName;
```
Comment (TZL0) [inline src/main/java/seedu/address/model/patient/NameContainsKeywordsPredicatePatient.java:9]: Is it better to name it without 'predicate' which may cause confusion?

File: src/main/java/seedu/address/model/patient/RemarkContainsKeywordsPredicate.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package seedu.address.model.patient;
+
+import java.util.function.Predicate;
+
+
+/**
+ * Tests that a {@code Patient}'s {@code Remark} matches any of the keywords given.
+ */
+public class RemarkContainsKeywordsPredicate implements Predicate<Patient> {
+    private final String predicateRemark;
```
Comment (TZL0) [inline src/main/java/seedu/address/model/patient/RemarkContainsKeywordsPredicate.java:10]: Same as NameContainsKeywordsPredicate

File: src/test/java/seedu/address/logic/commands/FindCommandTest.java [hunk 3]
```diff
@@ -1,29 +1,9 @@
...
-import seedu.address.model.ModelManager;
-import seedu.address.model.UserPrefs;
-import seedu.address.model.patient.NameContainsKeywordsPredicatePatient;
-
-/**
- * Contains integration tests (interaction with the Model) for {@code FindCommand}.
- */
-public class FilterNameCommandTest {
+/*
+public class FindCommandTest {
```
Comment (TZL0) [inline src/test/java/seedu/address/logic/commands/FindCommandTest.java:6]: Is it better to name this file FindPatientCommandTest?

File: src/test/java/seedu/address/model/ModelManagerTest.java [hunk 4]
```diff
@@ -159,8 +158,14 @@ public void equals() {
 
         // different filteredPatientList -> returns false
         String[] patientKeywords = ALICE.getName().fullName.split("\\s+");
+
+        String predicateName = patientKeywords[0];
```
Comment (TZL0) [inline src/test/java/seedu/address/model/ModelManagerTest.java:162]: Naming issue as above

File: src/main/java/seedu/address/model/patient/EmailContainsKeywordsPredicate.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * Tests that a {@code Patient}'s {@code Email} matches any of the keywords given.
+ */
+public class EmailContainsKeywordsPredicate implements Predicate<Patient> {
+
+    private final String predicateEmail;
+
+    public EmailContainsKeywordsPredicate(String predicateEmail) {
+        this.predicateEmail = predicateEmail;
```
Comment (TZL0) [inline src/main/java/seedu/address/model/patient/EmailContainsKeywordsPredicate.java:13]: Is it better to name it without 'predicate' which may cause confusion?

File: src/main/java/seedu/address/model/patient/AddressContainsKeywordsPredicate.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package seedu.address.model.patient;
+
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Patient}'s {@code Address} matches any of the keywords given.
+ */
+public class AddressContainsKeywordsPredicate implements Predicate<Patient> {
+    private final String predicateAddress;
```
Comment (TZL0) [inline src/main/java/seedu/address/model/patient/AddressContainsKeywordsPredicate.java:9]: Is it better to name it without 'predicate' which may cause confusion?

File: src/test/java/seedu/address/logic/commands/FindCommandTest.java [hunk 7]
```diff
@@ -33,14 +13,14 @@ public void equals() {
         NameContainsKeywordsPredicatePatient secondPredicate =
                 new NameContainsKeywordsPredicatePatient(Collections.singletonList("second"));
 
-        FilterNameCommand findFirstCommand = new FilterNameCommand(firstPredicate);
-        FilterNameCommand findSecondCommand = new FilterNameCommand(secondPredicate);
+        FindCommand findFirstCommand = new FindCommand(firstPredicate);
+        FindCommand findSecondCommand = new FindCommand(secondPredicate);
```
Comment (TZL0) [inline src/test/java/seedu/address/logic/commands/FindCommandTest.java:17]: Is it FindPatientCommand?

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 8]
```diff
@@ -63,13 +57,13 @@ public void parseCommand_exit() throws Exception {
...
     }
-    @Test
+    /*@Test
     public void parseCommand_filterName() throws Exception {
         List<String> keywords = Arrays.asList("foo", "bar", "baz");
-        FilterNameCommand command = (FilterNameCommand) parser.parseCommand(
-                FilterNameCommand.COMMAND_WORD + " " + keywords.stream().collect(Collectors.joining(" ")));
-        assertEquals(new FilterNameCommand(new NameContainsKeywordsPredicatePatient(keywords)), command);
-    }
+        FindCommand command = (FindCommand) parser.parseCommand(
```
Comment (TZL0) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:63]: FindPatientCommand?

**Review by TZL0 — CHANGES_REQUESTED** [review]
Hi, I have just found some naming issues, please check.

**Review by TZL0 — APPROVED** [review]

**Pull request merged**